### PR TITLE
Allow dex operation to be paused

### PR DESCRIPTION
**Pull request by jcompagni10**
Allow dex to be paused in the event of hack or other issues to ensure the rest of neutron chain can continue to run normally

### CODE REVIEW

**Comment by jcompagni10** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/9025944477

**Review by NicholasDotSol — APPROVED** [review]

**Review by pr0n00gler — CHANGES_REQUESTED** [review]
Please add a state migration which updates the DEX storage and particularly params state with added `paused` field

**Comment by jcompagni10** [comment]
> Please add a state migration which updates the DEX storage and particularly params state with added `paused` field

done

**Comment by jcompagni10** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/9084416344

**Review by swelf19 — COMMENTED** [review]
LGTM. But i do not see any tests at all. Could you add some.
1) You can pause
2) You can unpause
3) you can not perform any operation while dex is paused

Rerun integration tests pls, test run is failed.

**Review by swelf19 — COMMENTED** [review]
It's kinda strange we bump version for old params, doesn't it? Why cant we just a third param to the current params and do not create old v2

File: x/dex/keeper/migrations.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+type Migrator struct {
+	keeper Keeper
+}
+
+// NewMigrator returns a new Migrator.
+func NewMigrator(keeper Keeper) Migrator {
+	return Migrator{keeper: keeper}
+}
+
+// Migrate1to2 migrates from version 1 to 2.
```
Comment (joldie777) [inline x/dex/keeper/migrations.go]: Migrate2to3

**Review by joldie777 — CHANGES_REQUESTED** [review]

**Comment by jcompagni10** [comment]
> It's kinda strange we bump version for old params, doesn't it? Why cant we just a third param to the current params and do not create old v2

Perhaps I'm mistaken, but it seemed like bad practice to try and marshall the old params from the kvstore using the new proto definition. https://github.com/neutron-org/neutron/blob/feat%2Fpause_dex/x/dex/migrations/v3/store.go#L30

I played around and it does seem to work, but I still didn't feel comfortable taking the risk. @swelf19

**Comment by jcompagni10** [comment]
> LGTM. But i do not see any tests at all. Could you add some.
> 
> 1. You can pause
> 2. You can unpause
> 3. you can not perform any operation while dex is paused
> 
> Rerun integration tests pls, test run is failed.

done

**Comment by jcompagni10** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/9217868646

**Review by swelf19 — APPROVED** [review]

**Review by joldie777 — APPROVED** [review]

**Pull request merged**